src/mongocrypt-endpoint.c

```diff
@@ -77,13 +79,18 @@ _mongocrypt_endpoint_new (const char *endpoint_raw,
    prev = pos;
    pos = strstr (pos, ".");
    if (!pos) {
-      CLIENT_ERR (
-         "Invalid endpoint, expected dot separator in host, but got: %s",
-         endpoint->original);
-      goto fail;
+      if (!opts || !opts->allow_empty_subdomain) {
+         CLIENT_ERR (
+            "Invalid endpoint, expected dot separator in host, but got: %s",
+            endpoint->original);
+         goto fail;
+      }
+      /* OK, reset pos to the start of the host. */
+      pos = prev;
+   } else {
```

Comment:
nit: This is fine as is, but it almost leaves me feeling that the fail-case has priority here. There's nothing wrong with it, and it may be even preferable, but consider inverting the clause so that the "happy path" is primary.

Reply:
I think inverting helps readability. Done.